tdrs-backend/tdpservice/parsers/schema_defs/tribal_tanf/t7.py

```diff
@@ -0,0 +1,109 @@
+"""Schema for Tribal TANF T7 Row."""
+
+from tdpservice.parsers.util import SchemaManager
+from tdpservice.parsers.fields import Field, TransformField
+from tdpservice.parsers.row_schema import RowSchema
+from tdpservice.parsers.transforms import calendar_quarter_to_rpt_month_year
+from tdpservice.parsers import validators
+from tdpservice.search_indexes.models.tribal import Tribal_TANF_T7
+
+schemas = []
+
+validator_index = 7
+section_ind_index = 7
+stratum_index = 8
+families_index = 10
+for i in range(1, 31):
+    month_index = (i - 1) % 3
+    sub_item_labels = ["A", "B", "C"]
+    families_value_item_number = f"6{sub_item_labels[month_index]}"
+
+    schemas.append(
+        RowSchema(
+            model=Tribal_TANF_T7,
+            quiet_preparser_errors=i > 1,
+            preparsing_validators=[
+                validators.hasLength(247),
+                validators.notEmpty(0, 7),
+                validators.notEmpty(validator_index, validator_index + 24),
+            ],
+            postparsing_validators=[],
+            fields=[
+                Field(
+                    item="0",
+                    name="RecordType",
+                    friendly_name="record type",
+                    type="string",
+                    startIndex=0,
+                    endIndex=2,
+                    required=True,
+                    validators=[],
+                ),
+                Field(
+                    item="3",
+                    name="CALENDAR_QUARTER",
+                    friendly_name="calendar quarter",
+                    type="number",
+                    startIndex=2,
+                    endIndex=7,
+                    required=True,
+                    validators=[
+                        validators.dateYearIsLargerThan(1998),
+                        validators.quarterIsValid(),
```

Comment:
@elipe17 i discovered an issue with this validator. 

- If the quarter is a space, then no error is raised in the logs or the feedback report
- if the quarter = 0 or > 4, then this message is returned in the logs `Invalid quarter value.` and the processing status stays in the "pending" state which does not generate a feedback report. 

I believe this is an edge case, but does need to be addressed in a new ticket.

Reply:
#2807